packages/react/src/components/mantine-shared/Toolbar/ToolbarButton.tsx

```diff
@@ -33,6 +34,13 @@ export const ToolbarButton = forwardRef(
         {/*Creates an ActionIcon instead of a Button if only an icon is provided as content.*/}
         {props.children ? (
           <Button
+            // Needed as Safari doesn't focus button elements on mouse down
+            // unlike other browsers.
+            onMouseDown={() => {
+              if (buttonRef.current !== null) {
```

Comment:
Nice find!

I think you can get the element from the event (first argument of event handler). That way we don't need the ref / hacks and the code gets more succint